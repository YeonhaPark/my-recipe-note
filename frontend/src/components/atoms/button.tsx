import { ReactNode } from 'react';
import {
  Button as MButton,
  ButtonTypeMap,
  ButtonProps,
} from '@material-ui/core';
import { withStyles } from '@material-ui/styles';

export const IButton = <
  D extends React.ElementType = ButtonTypeMap['defaultComponent'],
  P = {},
>(
  props: ButtonProps<D, P>,
) => {
  return <MButton {...props} />;
};
interface Props {
  onClick?: () => void;
  children: string | ReactNode;
  variant?: 'text' | 'contained' | 'outlined';
  fullWidth?: boolean;
  style?: any;
  color?:
    | 'inherit'
    | 'primary'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning'
    | 'basic'
    | undefined;
}

const CustomButton = withStyles({
  root: {
    boxShadow: 'none',
  },
})(IButton);

export default function Button({
  onClick,
  children,
  color,
  variant,
  fullWidth,
  style,
}: Props): JSX.Element {
  return (
    <CustomButton
      onClick={onClick}
      color={color}
      style={style}
      variant={variant}
      fullWidth={fullWidth}
    >
      {children}
    </CustomButton>
  );
}

Button.defaultProps = {
  style: {},
  fullWidth: false,
  color: 'inherit',
  variant: 'text',
};
