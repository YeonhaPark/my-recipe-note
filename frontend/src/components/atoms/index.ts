import Button from './button';
import Chip, { ChipType, ChipColor } from './chip';
import Checkbox from './checkbox';

export { Button, Chip, Checkbox };
export type { ChipType, ChipColor };
