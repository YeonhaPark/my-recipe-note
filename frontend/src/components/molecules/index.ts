import Input from './input';
import Ingredient from './ingredient';
import IngredientType from './ingredient.type';
import Tags from './tags';

export { Input, Ingredient, Tags };
export type { IngredientType };
