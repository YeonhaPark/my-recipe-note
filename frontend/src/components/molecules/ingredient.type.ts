type IngredientType = {
  id: number;
  isChecked: boolean;
  name: string;
};

export default IngredientType;
