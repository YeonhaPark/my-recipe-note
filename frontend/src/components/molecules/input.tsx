/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx, css } from '@emotion/react';
import { Input as MInput, InputLabel } from '@material-ui/core';
import { RefObject } from 'react';

interface Props {
  myRef: React.Ref<string>;
  labelName: string;
  type: string;
  placeholder: string;
  required?: boolean;
}
export default function Input({
  myRef,
  labelName,
  type,
  placeholder,
  required,
}: Props) {
  return (
    <div
      css={css`
        margin-bottom: 1.5rem;
      `}
    >
      <InputLabel htmlFor={labelName}>{labelName}</InputLabel>
      <MInput
        inputRef={myRef}
        type={type}
        id={labelName}
        fullWidth
        placeholder={placeholder}
        required={required}
      />
    </div>
  );
}

Input.defaultProps = {
  required: false,
};
