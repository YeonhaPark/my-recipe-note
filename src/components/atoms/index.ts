import Button from './button';
import Chip from './chip';

export { Button, Chip };
