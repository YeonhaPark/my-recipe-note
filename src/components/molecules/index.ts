import Input from './input';
import Ingredient from './ingredient';

export { Input, Ingredient };
