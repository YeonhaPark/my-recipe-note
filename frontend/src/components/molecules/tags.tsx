/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx, css } from '@emotion/react';
import { Chip, ChipColor } from '../atoms';
import { Autocomplete, TextField } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { useState, useEffect } from 'react';
import { TagType } from '../../api/types';

const tagStyle = css`
  font-size: 0.75rem;
  flex: 0 1 1.5rem;
`;
const useStyles = makeStyles({
  root: {
    padding: '0.25rem 0',
  },
});

interface Props {
  tags: TagType[];
  setTags: React.Dispatch<React.SetStateAction<TagType[]>>;
}
const colors = [
  'primary',
  'secondary',
  'success',
  'warning',
  'default',
  'error',
  'info',
] as unknown as ChipColor;
export default function Tags({ tags, setTags }: Props) {
  const classes = useStyles();

  const [currentTagValue, setCurrentTagValue] = useState('');
  const [tagLabels, setTagLabels] = useState<string[]>(['']);

  // const addTag = (e: React.KeyboardEvent<HTMLInputElement>) => {
  //   if (e.key === 'Enter') {
  //     console.log('한번클릭했어여');
  //     const target = e.target as HTMLInputElement;
  //     const labelChipArr = tags.map((tag) => tag.label);
  //     if (!labelChipArr.includes(target.value)) {
  //       const randomColorIdx = Math.floor(Math.random() * colors.length);
  //       const randomColor: ChipColor = colors[randomColorIdx] as ChipColor;
  //       setTags([...tags, { label: target.value, color: randomColor }]);
  //     }
  //   }
  // }; 삭제해도될듯?

  const handleChange = (e: object, value: string[], reason: string) => {
    if (reason === 'clear') {
      setTags([]);
    }
    setTagLabels(value);
  };

  useEffect(() => {
    setTagLabels(tags.map((tag) => tag.label));
  }, [tags]);

  useEffect(() => {
    const event = new Event('change', {
      bubbles: true,
    });
    const node = document.getElementById('tags-standard');
    node && node.dispatchEvent(event);
  }, []);

  return (
    <div css={tagStyle}>
      <Autocomplete
        className={classes.root}
        onChange={handleChange}
        multiple
        id="tags-standard"
        options={tags.map((option) => option.label)}
        freeSolo
        value={tagLabels}
        renderTags={(value, getTagProps) => {
          const newTags = value.map((val: string, index: number) => {
            return {
              label: val,
              color: colors[index] as ChipColor,
            };
          });
          const chips = newTags.map((tag: TagType, index: number) => (
            <Chip
              {...getTagProps({ index })}
              key={tag.label}
              color={tag.color}
              variant="filled"
              label={tag.label}
            />
          ));
          return (
            <div
              css={css`
                margin: 0.25rem auto;
              `}
            >
              {chips}
            </div>
          );
        }}
        renderInput={(params) => {
          return (
            <TextField
              {...params}
              onChange={() => setCurrentTagValue}
              value={currentTagValue}
              variant="standard"
              label="Add tags"
            />
          );
        }}
      />
    </div>
  );
}
