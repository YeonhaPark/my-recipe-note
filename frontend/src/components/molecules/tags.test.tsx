import { mount } from 'enzyme';
import { Tags } from '../molecules';
import renderer from 'react-test-renderer';
import { ChipColor } from '../atoms';

const TagsProps = {
  tags: [
    {
      label: 'korean',
      color: 'info' as ChipColor,
    },
    {
      label: 'japanese',
      color: 'error' as ChipColor,
    },
  ],
  setTags: () => {},
};

const renderTags = () => mount(<Tags {...TagsProps} />);

describe('Tags', () => {
  it('renders Tags', () => {
    renderTags();
  });

  it('renders exact prop provided', () => {
    const tags = renderTags();

    expect(tags.prop('tags')).toEqual([
      {
        label: 'korean',
        color: 'info' as ChipColor,
      },
      {
        label: 'japanese',
        color: 'error' as ChipColor,
      },
    ]);
  });
});
