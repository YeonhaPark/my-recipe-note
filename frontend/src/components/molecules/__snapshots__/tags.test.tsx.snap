// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags matches snapshot 1`] = `
<div
  className="css-17i6ecd-Tags"
>
  <div
    aria-expanded={false}
    aria-owns={null}
    className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon makeStyles-root-1 css-1kpdewa-MuiAutocomplete-root"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor="tags-standard"
        id="tags-standard-label"
      >
        Add tags
      </label>
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-ghsjzk-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <div
          className="css-1hn6na0"
        >
          <span
            className="css-1loe067-CustomChip"
          >
            <div
              className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-tpwmjq-MuiChip-root"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                
              </span>
            </div>
          </span>
        </div>
        <input
          aria-activedescendant={null}
          aria-autocomplete="list"
          aria-controls={null}
          aria-invalid={false}
          autoCapitalize="none"
          autoComplete="off"
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="tags-standard"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          required={false}
          spellCheck="false"
          type="text"
          value=""
        />
        <div
          className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Clear"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
